packages/web/components/animation/dynamic-lottie-animation.tsx

```diff
@@ -0,0 +1,25 @@
+import dynamic from "next/dynamic";
+import { ComponentProps, FunctionComponent, useEffect, useState } from "react";
+
+const Lottie = dynamic(() => import("lottie-react"), { ssr: false });
+
+/** Loads the lottie libarary component, and it's
+ *  animation data dynamically (outside of the main bundle). */
+export const DynamicLottieAnimation: FunctionComponent<
+  {
+    /** JSON file source, likely a dynamic import: `import("./lottie.json")`. */
+    futureLottieContents: Promise<any>;
+  } & Omit<ComponentProps<typeof Lottie>, "animationData">
+> = (props) => {
+  // JSON contents
+  const [lottie, setLottie] = useState<any | null>(null);
+
+  // dynamic load JSON animation data
+  useEffect(() => {
+    if (!lottie) {
+      props.futureLottieContents.then(setLottie);
+    }
+  }, [lottie, props.futureLottieContents]);
+
+  return <Lottie {...props} animationData={lottie}></Lottie>;
```

Comment:
what happens if `null` is passed to `animationData`? `lottie` could be `null`

Reply:
I hardcoded it to test it out. It appears to fail gracefully and display nothing. Do you have an idea of how this should be handled?

Reply:
Good. I guess `Lottie` takes care of that for us. I was just going to suggest something like `{lottie} && <Lottie />` just to make sure we don't accidentally blow anything up.